### PR TITLE
Add Youth Activity Search Django and Wagtail application

### PR DESCRIPTION
**Pull request by cwdavies**
Bring into cfgov-refresh the wagtail application from teachers-digital-platform GitHub repository
Also crtool is referenced as a Django application and urls are updated.

## Additions

- cfgov/youth_activity_search

## Changes

- crtool in urls.py

## How to test this PR

1. tox

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing with the feature/section you're addressing, e.g., "Mega Menu: fix layout bug" or "Paying for College: Update content on Repay tool"
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/cfgov-refresh/) (edit the files in the `/docs` folder) – for basic, close-to-the-code documentation on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing

_[When new or significantly modified front-end functionality is present, the following things should be tested. Feel free to delete this section if not applicable to this PR.]_

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

[Further guidance on browser support](https://github.com/cfpb/development/blob/master/guides/browser-support.md)

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

**Comment by willbarton** [comment]
Could we name the Django app something other than `teachers_digital_platform` as we bring it? I think @Scotchester had some ideas.

**Comment by cwdavies** [comment]
> Could we name the Django app something other than `teachers_digital_platform` as we bring it? I think @Scotchester had some ideas.

@Scotchester Please share your ideas on a shorter name instead of `teachers_digital_platform`. I agree that we should rename the Django app. How about `youth_education` ?

**Comment by cwdavies** [comment]
Results from running tox after renaming teachers_digital_platform to youth_activity_search are as follows:
```
...
FAIL: test_deactivates_translations (core.tests.test_middleware.DeactivateTranslationsMiddlewareTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/DaviesC/Projects/CFPB/cfgov-refresh/cfgov/core/tests/test_middleware.py", line 231, in test_deactivates_translations
    self.assertEqual(translation.get_language(), 'en-us')
AssertionError: 'en' != 'en-us'
- en
+ en-us


----------------------------------------------------------------------
Ran 1614 tests in 121.692s

FAILED (failures=1, skipped=1)
```

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_facets_and_results.html [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+                        {% endfor %}
+                    </ul>
+                </fieldset>
+            </div>
+            {% endcall %}
+        {% endif %}
+        {% if facets.council_for_economic_education or facets.jump_start_coalition %}
+            </div>
+        {% endif %}
+        <input type="hidden" value="{% if page.results.search_query: %}{{ page.results.search_query }}{% endif %}" name="q">
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_facets_and_results.html]: Outdated

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_facets_and_results.html [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+{% endmacro %}
+
+<a class="skip-filters u-visually-hidden{% if not activities %} skip-filters__hidden{% endif %}" href="#tdp-search-results">Skip to search results</a>
+
+<aside class="content_sidebar content__flush-top-on-small content__flush-sides-on-small filters{% if not activities %} filters__hidden{% endif %}">
+    <h3 class="h2">Filter results by</h3>
+    <p class="u-hide-on-xs u-hide-on-sm">Narrow your search by selecting specific activity features.</p>
+    <form action="." method="get" data-js-hook="behavior_change-filter">
+        <input type="hidden" name="q" value="{% if search_query: %}{{ search_query }}{% endif %}">
+        <input type="hidden" name="page" inputmode="numeric" value="1">
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_facets_and_results.html]: Outdated

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_facets_and_results.html [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+    {% endfor %}
+{% endmacro %}
+
+<a class="skip-filters u-visually-hidden{% if not activities %} skip-filters__hidden{% endif %}" href="#tdp-search-results">Skip to search results</a>
+
+<aside class="content_sidebar content__flush-top-on-small content__flush-sides-on-small filters{% if not activities %} filters__hidden{% endif %}">
+    <h3 class="h2">Filter results by</h3>
+    <p class="u-hide-on-xs u-hide-on-sm">Narrow your search by selecting specific activity features.</p>
+    <form action="." method="get" data-js-hook="behavior_change-filter">
+        <input type="hidden" name="q" value="{% if search_query: %}{{ search_query }}{% endif %}">
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_search_facets_and_results.html]: Outdated

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_index_page.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                <div class="o-form__input-w-btn">
+                    <div class="o-form__input-w-btn_input-container">
+                        <div class="m-btn-inside-input input-contains-label">
+                            <label for="search-text" class="input-contains-label_before input-contains-label_before__search">
+                                {{ svg_icon('search') }}
+                                <span class="u-visually-hidden">The term to search for</span>
+                            </label>
+                            <input id="search-text" type="text" autocomplete="off" class="a-text-input" name="q"
+                                    placeholder="Enter your search term(s)"
+                                    value="{% if page.results.search_query: %}{{ page.results.search_query }}{% endif %}">
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_index_page.html]: Outdated

**Comment by cwdavies** [comment]
Created a new PR for review: https://github.com/cfpb/cfgov-refresh/pull/5815
It became very time consuming and difficult to rebase with master after the Django application was renamed from `teacher_digital_platform` to `youth_activity_search`

**Pull request closed without merging**